internal/controllers/gateway/httproute_controller.go

```diff
@@ -124,22 +125,38 @@ func (r *HTTPRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		)
 	}
 
-	// because of the additional burden of having to manage reference data-plane
-	// configurations for HTTPRoute objects in the underlying Kong Gateway, we
-	// simply reconcile ALL HTTPRoute objects. This allows us to drop the backend
-	// data-plane config for an HTTPRoute if it somehow becomes disconnected from
-	// a supported Gateway and GatewayClass.
-	return blder.For(&gatewayapi.HTTPRoute{}).
+	// We enqueue only routes that are:
+	// - attached during creation or deletion
+	// - have been attached or detached to a reconciled Gateway.
+	// This allows us to drop the backend data-plane config for a route if
+	// it somehow becomes disconnected from a supported Gateway and GatewayClass.
+	return blder.
```

Comment:
`blder`? Is it a typo -> `builder`
We have everywhere in the repo like this... 😅

Reply:
I guess that's named like this to not shadow the `https://github.com/Kong/kubernetes-ingress-controller/pull/6079/files#diff-57886002dd2fd6bfa8d355807bbd169962ca85034ef2cce7835a081b2a15c637R21` import.

Reply:
I see, I think we can do better, but id doesn't matter much

Reply:
Would you prefer us to change this? If so I suggest that this would be a separate PR. Otherwise, please resolve this thread.

Reply:
yes, it's out of scope here